tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -6637,9 +6661,17 @@ Status ConvertResize(OpConverterParams* params) {
 
   // Set layer parameters.
   layer->setResizeMode(resize_mode);
-  layer->setOutputDimensions(output_dimensions);
   layer->setAlignCorners(align_corners);
 
+  // Set output shape.
+  if (params->use_implicit_batch || HasStaticShape(inputs_dims)) {
+    // If all shape dims are known at build time, pass them as a parameter.
+    layer->setOutputDimensions(output_shape_dims);
+  } else {
+    // Otherwise, the output shape tensor becomes an extra input to the layer.
+    layer->setInput(1, *output_shape_tensor->trt_tensor());
+  }
+
```

Comment:
Let's merge this if-stmt with the previous if-stmt, by moving the code between the two if-stmts to before the first if-stmt.

Reply:
Actually, it's not possible to merge them, in case of dynamic shape, the concatenation of the shape tensor needs to be done before the resize layer itself, but then setting the output dimensions and/or adding the secondary input can only be done after the resize layer is created.

Reply:
I see what you meant. Can you (1) define "nvinfer1::IResizeLayer* layer = nullptr" before the if-stmt, and let the branches create the layer,  and (2) move the relevant code from [this block](https://github.com/tensorflow/tensorflow/blob/2825dbfe950cd514828da4bfed81a2402d779b35/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc#L6635-L6641) to after the if-stmt?

Reply:
I think I understand what you're asking, but I'm not sure this would result in cleaner code (unless I'm overlooking something obvious). The resize layer would still have to be created the same way on both branches -- basically because the `addResize()` method only takes a single input tensor anyway, so the second input would still need to be provided through a separate statement regardless. Please correct me if I missed the point of what you asked of course.

Reply:
Please let me know if there was anything else to do for this comment, as I think there's not much else that can be done with the way the layer is created in the current version.